core/object/callable_method_pointer.cpp

```diff
@@ -68,16 +68,19 @@ uint32_t CallableCustomMethodPointerBase::hash() const {
 	return h;
 }
 
-void CallableCustomMethodPointerBase::_setup(uint32_t *p_base_ptr, uint32_t p_ptr_size) {
+void CallableCustomMethodPointerBase::_setup(const char *p_base_ptr, uint32_t p_ptr_size) {
 	comp_ptr = p_base_ptr;
-	comp_size = p_ptr_size / 4;
+	comp_size = p_ptr_size;
+	h = 0;
 
 	// Precompute hash.
-	for (uint32_t i = 0; i < comp_size; i++) {
+	for (uint32_t i = 0; i < comp_size; i += sizeof(uint32_t)) {
+		uint32_t v = 0;
+		memcpy(&v, comp_ptr + i, MIN(sizeof(uint32_t), comp_size - i));
```

Comment:
I think it would be good to add a comment here to explain why we don't just cast to `const uint32_t *`